app/components/RandomQuote/RandomQuote.js

```diff
@@ -0,0 +1,117 @@
+// @flow
+
+import React, { Component } from 'react';
+import styles from './RandomQuote.css';
+import Button from '../Button';
+import type { QuoteEntity } from 'app/reducers/quotes';
+import Reaction from 'app/components/Reactions/Reaction';
+import Reactions from 'app/components/Reactions';
+import type { EmojiEntity } from 'app/reducers/emojis';
+import type { ID } from 'app/models';
+import { Link } from 'react-router';
+import { Flex } from 'app/components/Layout';
+
+type Props = {
+  fetchRandomQuote: () => Promise<Object>,
+  loggedIn: boolean,
+  className?: string,
+  addReaction: ({
+    emoji: string,
+    contentTarget: string
+  }) => Promise<*>,
+  deleteReaction: ({ reactionId: ID, contentTarget: string }) => Promise<*>,
+  fetchEmojis: () => Promise<*>,
+  fetchingEmojis: boolean,
+  emojis: Array<EmojiEntity>,
+  currentQuote: QuoteEntity
+};
+
+class RandomQuote extends Component<Props> {
+  render() {
+    const {
+      loggedIn,
+      className,
+      addReaction,
+      deleteReaction,
+      emojis,
+      fetchEmojis,
+      fetchingEmojis,
+      currentQuote
+    } = this.props;
+
+    let mappedEmojis = [];
+    if (!fetchingEmojis) {
+      mappedEmojis = emojis.map(emoji => {
+        const foundReaction = !currentQuote.reactionsGrouped
```

Comment:
This would be a bit more readable like this
```
const foundReaction = currentQuote.reactionsGrouped && currentQuote.reactionsGrouped.find(...)
```

Reply:
This ^, and similar at the other place this is done (QuotesRoute).

Reply:
I totally agree – not that it's really an excuse, but I just copied this code from Article Reactions. It feels like something that should be placed in reducers or something, and after a short discussion with @kotlarz I decided to just copy what he had for now instead of optimizing something we were planning to replace anyway. But it's exactly the kind of thing that noone ever gets around to fixing so I guess we might as well do it in this PR. 

In order to remove the need for this kind of code snippet every time someone wants to use reactions, where would you put this logic?

Reply:
If it has to run every time somebody adds/removes a reaction I also feel like it can be placed in some reducer or action.

Reply:
Yeah, putting it in reducer code works fine. 
You could also create a new `<LegoReactions>` (naming is hard, don't judge me) component that does everything between line 42-60 and 85-108. Then you can just pass the `reactionsGrouped` object into that component and the component does the mapping and rendering logic.

I think this makes the most sense if the `reactionsGrouped` looks the same across different entities (articles, quotes, etc.). It's probably also a bit simpler than trying to change the reducers a lot, and also a bit more flexible I guess.